File: Codeforces rounds/Codeforces rounds/Codeforces Round 874 (Div. 3)/A.cpp
// https://codeforces.com/contest/1833/problem/A

#include <bits/stdc++.h>

using namespace std;

typedef unsigned int uint;

#define fore(i, a, b) for (uint i = a, gmat = b; i < gmat; i++)

uint solve1(const string& s) {
  uint n = s.size();

  set<pair<char, char>> combs;

  fore(i, 0, n - 1) {
    combs.insert({s[i], s[i + 1]});
  }

  uint ans = combs.size();

  return ans;
}

int main(void) {
  ios_base::sync_with_stdio(false);
  cin.tie(NULL);

  uint t;
  cin >> t;

  fore(i, 0, t) {
    uint n;
    string s;
    cin >> n >> s;

    uint ans = solve1(s);
    cout << ans << '\n';
  }

  return EXIT_SUCCESS;
}

File: Codeforces rounds/Codeforces rounds/Codeforces Round 874 (Div. 3)/A.py
# https://codeforces.com/contest/1833/problem/A

def solve1(s: str) -> int:
    d: set[str] = set()
    for i in range(len(s)-1):
        d.add(s[i:i+2])
    return len(d)


def main():
    t: int = int(input())
    for _ in range(t):
        input()
        s: str = input()
        print(solve1(s))


if __name__ == '__main__':
    main()
